scripts/missions/windurst/1_1_The_Horutoto_Ruins_Experiment.lua

```diff
@@ -0,0 +1,393 @@
+        {
+            ['Mokyokyo'] = mission:progressEvent(118),
+
+            onEventFinish =
+            {
+                [118] = handleAcceptMission,
+            },
+        },
+
+        [xi.zone.WINDURST_WOODS] =
+        {
+            ['Rakoh_Buuma'] = mission:progressEvent(121),
+
+            onEventFinish =
+            {
+                [121] =  handleAcceptMission,
+            },
+        },
+    },
+
+    -- Mission has been accepted, these NPCs dialogue does not change during mission
+    {
+        check = function(player, currentMission)
+            return currentMission == mission.missionId
+        end,
+
+        [xi.zone.PORT_WINDURST] =
+        {
+            ['Janshura-Rashura'] =  mission:event(89),
+            ['Nine_of_Clubs'] = mission:event(86),
+            ['Puo_Rhen'] = mission:event(88),
+            ['Ten_of_Clubs'] = mission:event(87),
+        },
+
+        [xi.zone.WINDURST_WALLS] =
+        {
+            ['Keo-Koruo'] = mission:event(100),
+            ['Pakke-Pokke'] = mission:event(101),
+            ['Zokima-Rokima'] = mission:event(99),
+        },
+
+        [xi.zone.WINDURST_WATERS] =
+        {
+            ['Dagoza-Beruza'] = mission:event(121),
+            ['Mokyokyo'] = mission:event(123),
+            ['Panna-Donna'] = mission:event(122),
+            ['Ten_of_Hearts'] = mission:event(124),
+        },
+
+        [xi.zone.WINDURST_WOODS] =
+        {
+            ['Miiri-Wohri'] = mission:event(124),
+            ['Rakoh_Buuma'] = mission:event(127),
+            ['Sola_Jaab'] = mission:event(125),
+            ['Tih_Pikeh'] = mission:event(126),
+        },
+    },
+
+    -- Go to the Orastery in Port Windurst (Home Point #1) and talk to Hakkuru-Rinkuru (E-7) for a cutscene.
+    {
+        check = function(player, currentMission, missionStatus)
+            return currentMission == mission.missionId and missionStatus == 1
+        end,
+
+        [xi.zone.PORT_WINDURST] =
+        {
+            ['Hakkuru-Rinkuru'] = mission:progressEvent(90),
+
+            onEventFinish =
+            {
+                [90] = function(player, csid, option, npc)
+                    player:setMissionStatus(mission.areaId, 2)
+                end,
+            },
+        },
+    },
+
+    -- Head to Inner Horutoto Ruins at (J-7) and examine Gate: Magic Gizmo
+    {
+        check = function(player, currentMission, missionStatus)
+            return currentMission == mission.missionId and missionStatus == 2
+        end,
+
+        [xi.zone.EAST_SARUTABARUTA] =
+        {
+            ['Sama_Gohjima'] = mission:event(53),
+        },
+
+        [xi.zone.PORT_WINDURST] =
+        {
+            ['Hakkuru-Rinkuru'] = mission:event(91),
+        },
+
+        [xi.zone.INNER_HORUTOTO_RUINS] =
+        {
+             -- Gate: Magical Gizmo
+            ['_5c5'] =
+            {
+                onTrigger = function(player, npc)
+                    return mission:progressEvent(42)
+                end,
```

Comment:
Since there's only one function and no conditional, could simplify this to `['_5c5'] = mission:progressEvent(42),`